test/e2e/e2e.go

```diff
@@ -175,6 +191,64 @@ var _ = Describe("validation", func() {
 			})
 		})
 	})
+
+	Context("Creating AddressPool", func() {
+		It("Should create addresspool CR successfully", func() {
+			By("By creating a configmap")
+			addresspool := &metallbv1alpha.AddressPool{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "metallb.io/v1alpha1",
+					Kind:       "AddressPool",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      MetallbAddressPoolName,
+					Namespace: MetallbNameSpace,
+				},
+				Spec: metallbv1alpha.AddressPoolSpec{
+					Name:     "test1",
+					Protocol: "layer2",
+					Addresses: []string{
+						"1.1.1.1",
+						"1.1.1.100",
+					},
+				},
+			}
+
+			Expect(testclient.Client.Create(context.Background(), addresspool)).Should(Succeed())
+
+			key := types.NamespacedName{
+				Name:      MetallbAddressPoolName,
+				Namespace: MetallbNameSpace,
+			}
+
+			// Create addresspool resource
+			By("By checking AddressPool resource is created")
+			Eventually(func() error {
+				err := testclient.Client.Get(context.Background(), key, addresspool)
+				return err
+			}, timeout, interval).Should(Succeed())
+
+			// Create ConfigMap
+			By("By checking ConfigMap is created")
+			Eventually(func() error {
+				_, err := testclient.Client.ConfigMaps(MetallbNameSpace).Get(context.Background(), MetallbConfigMapName, metav1.GetOptions{})
+				return err
+			}, timeout, interval).Should(Succeed())
+
+			By("By Deleteing the ConfigMap")
```

Comment:
I think it would be more fitting to delete the addresspool first and validate that the configmap is also deleted (similar to what happens in AfterEach for MetalLBs)

Reply:
good point I will fix it